e2e/help/help.go

```diff
@@ -24,49 +24,54 @@ type ctx struct {
 	env e2e.TestEnv
 }
 
-var helpContentTests = []struct {
+var helpOciContentTests = []struct {
+	name string
 	cmds []string
 }{
 	// singularity oci
-	{[]string{"help", "oci"}},
-	{[]string{"help", "oci", "attach"}},
-	{[]string{"help", "oci", "create"}},
-	{[]string{"help", "oci", "delete"}},
-	{[]string{"help", "oci", "exec"}},
-	{[]string{"help", "oci", "kill"}},
-	{[]string{"help", "oci", "mount"}},
-	{[]string{"help", "oci", "pause"}},
-	{[]string{"help", "oci", "resume"}},
-	{[]string{"help", "oci", "run"}},
-	{[]string{"help", "oci", "start"}},
-	{[]string{"help", "oci", "state"}},
-	{[]string{"help", "oci", "umount"}},
-	{[]string{"help", "oci", "update"}},
+	{"HelpOci", []string{"help", "oci"}},
+	{"HelpOciAttach", []string{"help", "oci", "attach"}},
+	{"HelpOciCreate", []string{"help", "oci", "create"}},
+	{"HelpOciDelete", []string{"help", "oci", "delete"}},
+	{"HelpOciExec", []string{"help", "oci", "exec"}},
+	{"HelpOciKill", []string{"help", "oci", "kill"}},
+	{"HelpOciMount", []string{"help", "oci", "mount"}},
+	{"HelpOciPause", []string{"help", "oci", "pause"}},
+	{"HelpOciResume", []string{"help", "oci", "resume"}},
+	{"HelpOciRun", []string{"help", "oci", "run"}},
+	{"HelpOciStart", []string{"help", "oci", "start"}},
+	{"HelpOciState", []string{"help", "oci", "state"}},
+	{"HelpOciUmount", []string{"help", "oci", "umount"}},
+	{"HelpOciUpdate", []string{"help", "oci", "update"}},
 }
 
-func (c *ctx) testHelpContent(t *testing.T) {
-	for _, tc := range helpContentTests {
+func (c *ctx) testHelpOciContent(t *testing.T) {
+	for _, tc := range helpOciContentTests {
+
 		name := fmt.Sprintf("%s.txt", strings.Join(tc.cmds, "-"))
 
-		t.Run(name, func(t *testing.T) {
+		testHelpOciContentFn := func(t *testing.T, r *e2e.SingularityCmdResult) {
 			path := filepath.Join("help", name)
+			got := string(r.Stdout)
+			assert.Assert(t, golden.String(got, path))
+		}
 
-			c := exec.Command(c.env.CmdPath, tc.cmds...)
-
-			got := c.Run(t).Stdout()
+		e2e.RunSingularity(t, tc.name, e2e.WithArgs(tc.cmds...),
+			e2e.PostRun(func(t *testing.T) {
+				if t.Failed() {
+					t.Fatalf("Failed to run help command on test: %s", tc.name)
+				}
+			}),
+			e2e.ExpectExit(0, testHelpOciContentFn))
 
-			assert.Assert(t, golden.String(got, path))
-		})
 	}
 }
 
 func (c *ctx) testCommands(t *testing.T) {
-	tests := []struct {
+	testCommands := []struct {
 		name string
 		argv []string
```

Comment:
For readability, you could replace `argv []string` by `cmd string`, pass the `cmd` argument directly to `WithCommand(tt.cmd)` here https://github.com/sylabs/singularity/pull/3924/files#diff-2f7903baca57c225790dd52d14e1162eR115 and remove `WithArgs` (`WithCommand` split string argument by space so it's fine with `"instance start"` as command by example).
Then you could reuse `WithCommand(tt.cmd)` in `testCmdsFn` in conjunction with `WithArgs(tf.argv...)`